hack/start-redis-with-password.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# Default values for environment variables
+REDIS_PORT="${ARGOCD_E2E_REDIS_PORT:-6379}"
+REDIS_IMAGE_TAG=$(grep 'image: redis' manifests/base/redis/argocd-redis-deployment.yaml | cut -d':' -f3)
+
+# Check if ARGOCD_REDIS_LOCAL is 'true'
+if [ "$ARGOCD_REDIS_LOCAL" = 'true' ] && ! command -v redis-server &>/dev/null; then
```

Comment:
```suggestion
if [ "$ARGOCD_REDIS_LOCAL" = 'true' ]; then
    if ! command -v redis-server &>/dev/null; then
        echo "Redis server is not installed locally. Please install Redis or set ARGOCD_REDIS_LOCAL to false."
        exit 1
    fi

    # Start local Redis server with or without password
```

if the env is set to true and the command is installed locally, the local redis server should be used. However, the `exit 1`  statement makes the following code block for starting the local Redis server unreachable.